engine/src/org/pentaho/di/trans/steps/databaselookup/DatabaseLookup.java

```diff
@@ -640,6 +479,60 @@ private void loadAllTableDataIntoTheCache() throws KettleException {
     }
   }
 
+  private void putToDefaultCache( Database db, List<Object[]> rows ) {
+    RowMetaInterface returnRowMeta = db.getReturnRowMeta();
+    // Copy the data into 2 parts: key and value...
+    //
+    for ( Object[] row : rows ) {
+      int index = 0;
+      RowMeta keyMeta = new RowMeta();
```

Comment:
@akhayrutdinov please consider using RowMeta.clone() as it should now be much faster than adding individual value metas (https://github.com/pentaho/pentaho-kettle/commit/88e0ebbfb0caa67c5f72d19066697d1fa1856793)